src/WorkBC.Shared/Services/GeocodingCachingService.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using WorkBC.Data;
+using WorkBC.Data.Model.JobBoard;
+
+namespace WorkBC.Shared.Services
+{
+    public class GeocodingCachingService : IGeocodingService
+    {
+        private readonly JobBoardContext _dbContext;
+        private readonly ILogger<IGeocodingService> _logger;
+        private readonly IGeocodingService _geocodingService;
+
+        public GeocodingCachingService(JobBoardContext dbContext, IGeocodingService geocodingService,
+            ILogger<IGeocodingService> logger = null)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+            _geocodingService = geocodingService;
+        }
+
+        public async Task<GeocodedLocationCache> GetLocation(string location)
+        {
+            GeocodedLocationCache cachedLocation =
+                await _dbContext.GeocodedLocationCache.FirstOrDefaultAsync(g => g.Name == location);
```

Comment:
Based on the new unit tests, the comparison of `g.Name == location` is case-insensitive